packages/flutter_test/lib/src/test_text_input.dart

```diff
@@ -21,24 +21,39 @@ const String _kTextInputClientChannel = 'flutter/textinputclient';
 /// * [WidgetTester.showKeyboard], which uses this class to simulate showing the
 ///   popup keyboard and initializing its text.
 class TestTextInput {
+  /// Installs this object as a mock handler for [SystemChannels.textInput].
   void register() {
-    SystemChannels.textInput.setMockMethodCallHandler(handleTextInputCall);
+    SystemChannels.textInput.setMockMethodCallHandler(_handleTextInputCall);
   }
 
   int _client = 0;
   Map<String, dynamic> editingState;
 
-  Future<dynamic> handleTextInputCall(MethodCall methodCall) async {
+  Future<dynamic> _handleTextInputCall(MethodCall methodCall) async {
     switch (methodCall.method) {
       case 'TextInput.setClient':
         _client = methodCall.arguments[0];
         break;
+      case 'TextInput.clearClient':
+        _client = 0;
+        _isVisible = false;
+        break;
       case 'TextInput.setEditingState':
         editingState = methodCall.arguments;
         break;
+      case 'TextInput.show':
+        _isVisible = true;
+        break;
+      case 'TextInput.hide':
+        _isVisible = false;
+        break;
     }
   }
 
+  bool get isVisible => _isVisible;
```

Comment:
add docs to this one in particular, since it's not obvious what is visible at first glance